### ISSUE TITLE
Properly display unit informations in graphs

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#7845]
The International Bureau of Weights and Measures (BIPM) makes the following recommendation for displaying unit information in graphs and tables (see [SI Brochure](https://www.bipm.org/documents/20126/41483022/SI-Brochure-9-EN.pdf/2d2b50bf-f2b4-9661-f402-5f9d66e4b507?version=1.9&download=true), pag. 148-149):

> Symbols for units are treated as mathematical entities. In expressing the value of a quantity as the product of a numerical value and a unit, both the numerical value and the unit may be treated by the ordinary rules of algebra. This procedure is described as the use of quantity calculus, or the algebra of quantities. For example, the equation p = 48 kPa may equally be written as p/kPa = 48.It is common practice to write the quotient of a quantity and a unit in this way for a column heading in a table, so that the entries in the table are simply numbers. (...) The axes of a graph may also be labelled in this way, so that the tick marks are labelled only with numbers, as in the graph below

For example, the caption for a lenght l would be l/m, while the caption for a current i would be i/A.

I would suggest we adopt this convention for legends and axes captions in OMEdit and OMPlot, instead of the current practice of putting the unit in parentheses.

@ceraolo, what do you think?

**Comment by max-privato** [comment]
If I remember well we discussed this already a few yeas ago. At that time the legend of names in Openmodelica carried square brackets, e.g.: 
diode.i [A]

I did not like the square brackets since CGPM requires a different use for them, indicating the units of measure of a quantity:
[diode.i]=A

If I remember well I mentioned that the CGPM required to write
diode.i/A

but I did not dare to ask to make the big leap, so as a compromise I proposed 
diode.i (A)

as it is now.

So your proposal finds me already prepared on the topic. I'm not sure what to recommend.  Using diode.i/A is so conceptually powerful that it tempts. In practice, it recalls us that the number indicating a quantity is always dimensionless since it is the ratio of the physical quantity and its unit which clarifies things a lot.

However, it does not meet the favour of many. Don't know if because it disrupts consolidated habits or what.

Should we dare and make the leap? Not sure, I'm not in favour and not against. Maybe we should get more opinions on this.

I just mention that if we adopt this we will have some indications that appear at bit tricky.  Consider for instance a current density diode.j. It is measured in A/mm2.
How would do write on the axis? I presume it should be 
diode.j/(A/mm2).

**Comment by casella** [comment]
> How should we write it on the axis legend? I presume this way:
> diode.j/(A/mm2).

Absolutely.

**Comment by casella** [comment]
@max-privato I remember we had this discussion some time ago but couldn't trace it online. I got definitely hooked by this formalism in the meantime.

Maybe we should hear other opinions. @rfranke, @niklwors, @dietmarw, what do you think?

**Comment by rfranke** [comment]
Fine with me. I would just add spaces to better separate the name from the unit, e.g.:
 diode.i / A
 diode.j / (A/mm2)

**Comment by dietmarw** [comment]
My personal preference is still `[unit]` since this way it is hard to confuse equations with quantities and their units. As far as I can see the BIPM does not demand the quotient notation but is simply recommending a convention. In this case I think they are recommending something that makes the interpretation harder to read for end users.

**Comment by max-privato** [comment]
As I said in my previous comment [unit] is against BIPM, even though very common.
I like the vocation of OM to never go against standards.  So, as I said, I'm not in favour nor against adopting the BIPM convention proposed in this ticket in OM, but I'm strongly against getting back to [unit].

**Comment by dietmarw** [comment]
Well in this case it is not a violation but simply not following a suggested convention. There is a difference between standards and recommendation. And in this case I think making it more user friendly and less confusing might be enough reason to not follow the recommendation.  But that's just my opinion which I was asked for by @casella , so here it is ;-)

**Comment by max-privato** [comment]
Yes, sure, @dietmarw . In my previous comment I just discussed the use of square brackets instead of round ones.

**Comment by casella** [comment]
I agree with the comments, I guess providing both options would be the optimal solution, what do you think? Then of course we can fight over which should be the default one :P

**Comment by ceraolo** [comment]
Just a very minor glitch: I would drop "s" in "informations" in this ticket's title
